buildtools/maven/maven.go

```diff
@@ -59,53 +72,154 @@ func (m *Maven) Compile(dir string) error {
 	return err
 }
 
-func Modules(dir string) ([]string, error) {
+// CachedManifest returns the manifest from the internal cache, or nil if it is not set.
+func (m *Maven) CachedManifest(target string) *Manifest {
+	if m.manifestCache == nil {
```

Comment:
[nit] This nil check isn't required, because indexing into a nil map will already return nil here

Reply:
thanks—I'm always forgetting whether it's just writing to or reading/writing to nil maps that's not allowed